### PR TITLE
test

### PR DESCRIPTION
**Pull request by GRANT225**

### CODE REVIEW

**Comment by jrome5** [comment]
"System.NullReferenceException: Object reference not set to an instance of an object
  at Packages.Rider.Editor.UnitTesting.CallbackInitializer..cctor () [0x00001] in C:\Users\GRANT\Documents\maze-game-master\maze-game-master\mazegame\Library\PackageCache\com.unity.ide.rider@1.1.4\Rider\Editor\UnitTesting\CallbackInitializer.cs:13 
UnityEditor.EditorAssemblies:ProcessInitializeOnLoadAttributes(Type[])"

There seems to be an issue when i clone this branch. Not sure what seems to try and link to but it seems to want to grab an address from your documents.

**Comment by GRANT225** [comment]
To view maze-game.zip, sign in<https://heriotwatt-my.sharepoint.com/personal/gms4_hw_ac_uk/_layouts/15/acceptinvite.aspx?invitation=%7BACF08DF8%2DDA3F%2D4EEB%2DBB0F%2DF231AD424E22%7D&listId=ac5621c8%2D3bd1%2D4637%2Dbd78%2D1e739f1c5e19&itemId=ec66c34a%2Deb32%2D472e%2Db9b4%2D1d237a1da18f> or create an account.
________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by GRANT225** [comment]
Swanson, Grant M has shared a OneDrive for Business file with you. To view it, click the link below.
<https://heriotwatt-my.sharepoint.com/personal/gms4_hw_ac_uk/Documents/Attachments/maze-game.zip>
[https://r1.res.office365.com/owa/prem/images/dc-zip_20.png]<https://heriotwatt-my.sharepoint.com/personal/gms4_hw_ac_uk/Documents/Attachments/maze-game.zip>
maze-game.zip<https://heriotwatt-my.sharepoint.com/personal/gms4_hw_ac_uk/Documents/Attachments/maze-game.zip>


________________________________

File: mazegame/Assets/Scripts/Menu/MenuText.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using UnityEngine.EventSystems;
+using UnityEngine.SceneManagement;
+
+public class MenuText : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler, IPointerDownHandler
+{
+    
+
+    public void OnPointerEnter(PointerEventData eventData)
+    {
+        this.transform.localScale += new Vector3(0.4f, 0.4f, 0.4f);
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Menu/MenuText.cs]: please assign this to a private class variable. Such as `private Vector3 text_scaling = Vector3(0.4f, 0.4f, 0.4f);`

File: mazegame/Assets/Scripts/Menu/MenuText.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public void OnPointerExit(PointerEventData eventData)
+    {
+        this.transform.localScale -= new Vector3(0.4f, 0.4f, 0.4f);
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Menu/MenuText.cs]: same here. add class variable. `this.transform.localScale -= text_scaling

File: mazegame/Assets/Scripts/Maze/TextEditor.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class TextEditor : MonoBehaviour
+{
+
+    string key1_found = "You found a key! I wonder where it fits...";
+    string key2_found = "You found a second key!";
+    string key3_found = "You found a third key! ... *click*... Sounds like a door has opened";
```
Comment (jrome5) [inline mazegame/Assets/Scripts/Maze/TextEditor.cs]: From what i can tell, these class member variables are never used anywhere else apart from in one instance inside the class. 
1. Move these to `Start` function
2. Make them type `var`
3. Avoid the use of numbers in variable names. Please rename to `first_key_found`

**Review by jrome5 — CHANGES_REQUESTED** [review]

**Pull request merged**